### PR TITLE
Add tests and error handling for mergePackageJson and prettyPrintMarkdown

### PR DESCRIPTION
**Pull request by ecraig12345**

### CODE REVIEW

File: packages/just-scripts-utils/src/mergePackageJson.ts [hunk 1]
```diff
@@ -40,3 +39,29 @@ export function mergePackageJson(original: IPackageJson, incoming: IPackageJson)
...
 }
+
+/**
+ * Exported for testing only. Determines whether a dep should be updated.
+ * Returns false if either version includes semver elements we don't understand
+ * (any characters besides number and ~ ^).
+ * @param originalVersion The original dep version (default 0.0.0)
+ * @param incomingVersion The incoming dep version
+ */
+export function _shouldUpdateDep(
```
Comment (kenotron) [inline packages/just-scripts-utils/src/mergePackageJson.ts]: I like this convention of a "private" exported function using _ to start functions.

File: packages/just-scripts-utils/src/__tests__/mergePackageJson.spec.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import { mergePackageJson, _shouldUpdateDep } from '../mergePackageJson';
+import { IPackageJson } from '../IPackageJson';
+
+describe('_shouldUpdateDep', () => {
+  it('correctly detects deps needing update', () => {
+    // okay if old version is undefined
+    expect(_shouldUpdateDep(undefined, '0.0.1')).toBe(true);
```
Comment (kenotron) [inline packages/just-scripts-utils/src/__tests__/mergePackageJson.spec.ts]: Can you split these into smaller `it()` test cases? It is relatively difficult to parse test failures this way.

Comment (ecraig12345) [reply packages/just-scripts-utils/src/__tests__/mergePackageJson.spec.ts]: How small? Like I mentioned on another PR, Jest does do a good job of pointing out which line failed.

Comment (ecraig12345) [reply packages/just-scripts-utils/src/__tests__/mergePackageJson.spec.ts]: Split somewhat. What do you think?

Comment (kenotron) [reply packages/just-scripts-utils/src/__tests__/mergePackageJson.spec.ts]: great!

**Pull request merged**